rpcserver.go

```diff
@@ -4615,30 +4615,16 @@ func (r *rpcServer) UpdateChannelPolicy(ctx context.Context,
 	// With the scope resolved, we'll now send this to the
 	// AuthenticatedGossiper so it can propagate the new policy for our
 	// target channel(s).
-	err := r.server.authGossiper.PropagateChanPolicyUpdate(
+	chanPolicies, err := r.server.authGossiper.PropagateChanPolicyUpdate(
 		chanPolicy, targetChans...,
 	)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
What happens if there are concurrent fee update calls, say call A and call B, for the same channel? I think it is possible that A updates the db, then B updates the db, then B applies policy B to the link and finally A applies policy A to the link. At that point the db and link are no longer in sync.

Reply:
Looks like a pre-existing issue, but it seems unlikely given that the gossiper can only handle one `PropagateChanPolicyUpdate` call at a time? B would start after A is done, at which point A is likely to have reached the switch already.

Reply:
Yes, indeed looks unlikely, but it still is a small gap. And pre-existing indeed. Maybe we can just throw a lock around it all? Or at least add a TODO describing this. Or do nothing at all, given that it is pre-existing.

Reply:
Added a lock around the db+link update code!